tests/integration/security/reachability/reachability_test.go

```diff
@@ -44,280 +43,192 @@ import (
 func TestReachability(t *testing.T) {
 	framework.NewTest(t).
 		Run(func(ctx framework.TestContext) {
-			istioCfg := istio.DefaultConfigOrFail(t, ctx)
+			istioCfg := istio.DefaultConfigOrFail(ctx, ctx)
 
-			systemNS := namespace.ClaimOrFail(t, ctx, istioCfg.SystemNamespace)
-			ns := namespace.NewOrFail(t, ctx, "reachability", true)
+			systemNS := namespace.ClaimOrFail(ctx, ctx, istioCfg.SystemNamespace)
+			ns := namespace.NewOrFail(ctx, ctx, "reachability", true)
 
-			ports := []echo.Port{
+			var a, b, headless, naked echo.Instance
+			echoboot.NewBuilderOrFail(ctx, ctx).
+				With(&a, config("a", ns, false, nil)).
+				With(&b, config("b", ns, false, nil)).
+				With(&headless, config("headless", ns, true, nil)).
+				With(&naked, config("naked", ns, false, echo.NewAnnotations().
+					SetBool(echo.SidecarInject, false))).
+				BuildOrFail(ctx)
+
+			callOptions := []echo.CallOptions{
+				{
+					PortName: "http",
+					Scheme:   scheme.HTTP,
+				},
 				{
-					Name:     "http",
-					Protocol: model.ProtocolHTTP,
+					PortName: "http",
+					Scheme:   scheme.WebSocket,
```

Comment:
Why are we using scheme.WebSocket for http (as opposed to scheme.HTTP) in this case?

Reply:
We're testing the websocket protocol (which is over http).

Reply:
But does this test needs websocket or just http? Also, if we're testing websocket protocol, shouldn't we have a different test for the test framework?

Reply:
It's probably not entirely necessary, but it preserves the tests from the old pilot e2e tests.